### ISSUE TITLE
[Bug]: ERR_GEN_INVALID error while updating cache

### ISSUE THREAD

**Issue by alyxme** [Yimura/YimMenu#2107]
### Describe the bug

Latest build insist to update cache but fails while updating.

### Steps To Reproduce

Build latest or download release
Try update
...
App crash

### Expected Behavior

Expect update cache without crashing app

### Logs and/or screenshots

<details><summary>Details</summary>
<p>

```
[16:25:08.3227853][INFO/backend.cpp:187] Starting script: World
[16:25:08.8108019][INFO/tunables_service.cpp:85] Saving 34386 tunables to cache
[16:25:09.1867200][INFO/tunables_service.cpp:36] Loading tunables from cache
[16:25:32.6934544][INFO/gta_data_service.cpp:264] Rebuilding cache started...
[16:25:32.6936931][DEBUG/yim_fipackfile.cpp:125] GTA install directory: D:\Games\GTAV
[16:25:32.7654425][WARN/log_message_error_box.cpp:37] Error Code: ERR_GEN_INVALID

```
</p>
</details> 

### Store

Epic Games

### Additional context

Tried: 
Reinstall R* Launcher with completely removing profile from Documents and Program Files,
Removing Yim dir from Roaming,
Reinstall launcher to gamedir,
Start in SP or MP,
Launch as Admin.

No spaces in game dir

In cache dir files generated:
sc_pointers.bin
gta_pointers.bin
tunables.bin (after clean start)

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

**Comment by Yimura** [comment]
How did you previously generate your cache?

I made an update to the GTA data service in YimMenu to fix some issues with crash models being cached.
Because of this you have to update the cache if you have an older version of the cache.

**Comment by alyxme** [comment]
Previously I updated after this [95f1db0] and never after. Also I noticed few release earlier when I tried to update, same thing happen but since it never insist to update a cache I ignored it.

**Comment by Yimura** [comment]
Do you have mods installed or have a modded game installation?

**Comment by alyxme** [comment]
No one and never. Installed clean from Epic on W'11

**Comment by Yimura** [comment]
What's your game's language set to? Do you have the game installed on a hard drive?

I find it odd that you were previously able to generate a cache with [95f1db0] but not after.
I'll take a look comparing the changes made between then and now.

**Comment by alyxme** [comment]
Game is set to English locale but installed on harddrive. System is on SSD.

**Comment by alyxme** [comment]
Tried to rebuild with yim_fipackfile.cpp from commit [95f1db0] but unsuccessful.

**Comment by xiaoxiao921** [comment]
ERR_GEN_INVALID errors are because you have mods installed in your game folder and our code does not properly handle it

**Comment by alyxme** [comment]
I dont have any mods installed and never had them! Never modified RPFs, don't have any custom folders or dlls in gamedir, nothing but clean install. As always, truth is out there. Hope to find it.
Last time, with your kind help, unexpected Launcher.rpf was skipped but for now I reinstalled SC with wiping profile from appdata but it doesnt help at all.
I can't use newer builds that insists to rebuild that unavoidable and so much important cache. 
It would be much useful if we have more info in the logs about what exactly caused crash like that.

**Comment by Yimura** [comment]
Just as a confirmation, if you inject [95f1db0] it will generate a cache correctly?

Here's a comparison I've done with your commit hash against master:
https://github.com/YimMenu/YimMenu/compare/95f1db08947581e269c3d95720a9fefdcafff55b...master

I'll check again but yesterday when I looked at it I didn't find anything that could lead to it working wildly differently.

**Comment by xiaoxiao921** [comment]
> I dont have any mods installed and never had them! Never modified RPFs, don't have any custom folders or dlls in gamedir, nothing but clean install. As always, truth is out there. Hope to find it. Last time, with your kind help, unexpected Launcher.rpf was skipped but for now I reinstalled SC with wiping profile from appdata but it doesnt help at all. I can't use newer builds that insists to rebuild that unavoidable and so much important cache. It would be much useful if we have more info in the logs about what exactly caused crash like that.

You are not providing a complete log so that would be a nice start

**Comment by Yimura** [comment]
As a temporary solution you can use the cache I generated using my system but I'd still like to debug the issue with you since we're having issues reproducing this ourselves.
[cache.zip](https://github.com/YimMenu/YimMenu/files/12568490/cache.zip)

Could you try using the build from #1859, it will not fix your issue but might give us some more insights on what's going wrong.

**Comment by alyxme** [comment]
I very apreciate how both of you trying do the best! So this is an example for me to try do the best too!
I used yim_fipackfile.cpp from [95f1db0] and put it in latest with rebuild. - crash
Tried to inject dll generated by Bot from commit above but it crash even while loading.
Version from 1859 also crashed while updating cache.
With pregenerated cache from zip is also insists to update cache - crash
I'm apologise for providing not full log because I thought only last lines are important. There is full [cout.log](https://github.com/YimMenu/YimMenu/files/12568666/cout.log) from latest build.
What else I should do to help to avoid this issue?

**Comment by Yimura** [comment]
It even insist on updating when using the cache generated on my machine? That's extremely odd and shouldn't happen.
I'll make some changes to #1859 to log some more information... I'll let you know when I've done that.

You can confirm you placed the files from the zip under `%appdata%/YimMenu/cache/`?

**Comment by alyxme** [comment]
Yes ofcourse I placed in this place and backuped existing by renaming old [cache](https://github.com/YimMenu/YimMenu/files/12568964/cac123he.zip) folder.

**Comment by alyxme** [comment]
I used build from Release by github bot and that build has loaded with cache from zip above but unexpected crashes at random point: first while loading, and next when game fully loaded and after menu is opened. I removed work dir (with backup) and tried again to start with rebuild but crashed again ([cout.log](https://github.com/YimMenu/YimMenu/files/12569530/cout.log)). Super odd. Previous build was very stable and never crashed at random points. Tried to load on old W7 from other disk, launcher has autoupdated but game never starts because R* cant connect to its servers. Hope to solve this issue.

**Comment by Yimura** [comment]
@alyxme, I've added some extra logging to #1859 could you try and run that build again (I'm sure it'll crash but mostly looking forward to the logs)?

Also I've noticed your cout.log file ended with:
```
[17:14:01.7356793][DEBUG/yim_fipackfile.cpp:125] GTA install directory: D:\Games\GTAV
[17:14:01.8041807][WARN/log_message_error_box.cpp:37] Error Code: ERR_GEN_INVALID
```
Is there anything below this or is that all before the game dies?

**Comment by alyxme** [comment]
Thank you! I will make test today later. The log file is full and nothing is below `[17:14:01.8041807][WARN/log_message_error_box.cpp:37] Error Code: ERR_GEN_INVALID`. "OK error" popup did not appeared.

**Comment by alyxme** [comment]
The [cout.log](https://github.com/YimMenu/YimMenu/files/12586376/cout.log) from 1859. Still same crash while updating cache. 
Only gta_pointers.bin and sc_pointers.bin are generated.
EDIT: I cleaned yim folder from appdata before test, so log above is from clean boot.
EDIT2: If it helps: [GTA5.exeheader.txt](https://github.com/YimMenu/YimMenu/files/12586452/GTA5.exeheader.txt)
EDIT3: Game [dir.txt](https://github.com/YimMenu/YimMenu/files/12586554/dir.txt)

**Comment by gir489returns** [comment]
> Thank you! I will make test today later. The log file is full and nothing is below `[17:14:01.8041807][WARN/log_message_error_box.cpp:37] Error Code: ERR_GEN_INVALID`. "OK error" popup did not appeared.

This is probably a black box error from how we call the function to get the information from the RPF files.

**Comment by alyxme** [comment]
I did log verbose to yim_fipackfile.cpp and cache_file.cpp: [gta_data.zip](https://github.com/YimMenu/YimMenu/files/12589067/gta_data.zip) resulting [cout.log](https://github.com/YimMenu/YimMenu/files/12589070/cout.log). Noticed peds.bin and vehicles.bin not presented in yimdir but tried to read that not existed files. Also vehicles.meta is last readed file.

**Comment by xiaoxiao921** [comment]
I checked again and ERR_GEN_INVALID errors only happens when your rpfs are not in the expected format

RPF_ENCRYPTION_AES  0x0FFFFFF9
RPF_ENCRYPTION_NG   0x0FEFFFFF

in IDA search for `BE0E0AEA`

**Comment by gir489returns** [comment]
> I checked again and ERR_GEN_INVALID errors only happens when your rpfs are not in the expected format
> 
> RPF_ENCRYPTION_AES 0x0FFFFFF9 RPF_ENCRYPTION_NG 0x0FEFFFFF
> 
> in IDA search for `BE0E0AEA`

Could the source of the RPF files matter? @alyxme which store did you acquire GTA V from? Rockstar, Steam or Epic?

**Comment by alyxme** [comment]
They are from Epic but installed on W7. I migrated to W11 in this year. I just added installdir to Epic and it says ok after parcing it and even installed some updates after. Previously I used to look at gameworld with CodeWalker. Not sure if it caused files to modified. I know about OpenIV and mods to use with it but dont wish to mess with it and never tried. I will look to mentioned hash with IDA later. Im not sure if I only one with that error but atleast I gave a feedback.

**Comment by xiaoxiao921** [comment]
I already told you, your game installation is not clean, I said this 2 days ago already

**Comment by alyxme** [comment]
And reinstalling of 120Gb+ will take fewer time than 2 days. XD

**Comment by gir489returns** [comment]
> And reinstalling of 120Gb+ will take fewer time than 2 days. XD

We're not going to compensate for your garbage environment. Sounds like a (You) issue.

**Comment by alyxme** [comment]
Reinstalling did not fixed the issue with yim cache.

**Comment by xiaoxiao921** [comment]
https://www.epicgames.com/help/en-US/epic-games-store-c5719341124379/launcher-support-c5719357217435/how-do-i-verify-game-files-in-the-epic-games-launcher-a5720337301659

**Comment by alyxme** [comment]
I installed the game from Epic and it successfully verified installation. Should I do that again? Im sure it will show 100% correct consistency of gamefiles. Also I did chkdsk with no errors.

**Comment by alyxme** [comment]
I did veryfing with Epic again. After 100% completion there no additional info about how it results. After that completion I starting again without yimdir from appdata but unfortunately issue is still exist. [cout.log](https://github.com/YimMenu/YimMenu/files/12600189/cout.log). 
So. Im sure about gamefiles is 100% pure and clean, R* Launcher freshly reinstalled with SC as well. With kind Yimura's cache.zip appcrash appear while loading and cannot even go further. Older build loads with old cache absolutely fine! It drives me mad about. Why this cache is so important, should it be simplified to ignore that unavoidable generation? No. Atleast for now.

**Comment by alyxme** [comment]
I found the reason and fixed this issue. I was right about pureness of gamefiles, they are all clear and non modified. The reason why all this happened was the last thing I paid attention to.

**Comment by Yimura** [comment]
Paid attention to?

**Comment by alyxme** [comment]
Settings of this injector was the issue!
<details><summary>Details</summary>
<p>

![issue](https://github.com/YimMenu/YimMenu/assets/62597693/dd5eeb45-c0ab-4e61-985d-bcf695fff31e)

</p>
</details> 

 With default settings there no problem with cache generation. oof!

**Comment by alyxme** [comment]
PS. [cout.log](https://github.com/YimMenu/YimMenu/files/12601543/cout.log)

**Comment by Yimura** [comment]
And what setting were you originally using?

**Comment by alyxme** [comment]
All above caused by the settings at the screenshot. I always used them before as recomended (for gta5) to safely use injection as undetected but they interfered with cache generation.

**Comment by alyxme** [comment]
> PS. [cout.log](https://github.com/YimMenu/YimMenu/files/12601543/cout.log)

But some exceptions still in the log. Atleast there was no crash. Glad I did that build with additional log verbose. Hope it helps to tune it and other things too. Thanks yall!~

**Comment by Yimura** [comment]
I use the most default injectors and have had no issues. Custom injector settings being safer is a MYTH mostly spread by YouTubers who claimed that from other games.

**Comment by Yimura** [comment]
@alyxme I see in  your log files that you have got custom DLC installed which is why that error was popping up:
```
[19:22:42.0538188][INFO/yim_fipackfile.cpp:44] Failed opening Resources\XMLs\bb6fdd-FIB Mansion\FIB Mansion\custom_maps\dlc.rpf
```

Your custom injector settings where preventing YimMenu from being able to log the error properly which is why we were never able to see the custom DLC in your game folder.

**Comment by alyxme** [comment]
I did settings to injector as recomended by someone from gta-related forum because "gta-antisig.dll" was a thing some time ago. Also I did not install any of mods! This is pure "vanilla" (Premium Edition) version clean installed from Epic as it is few hours ago!

**Comment by gir489returns** [comment]
I have a feeling you have GTA V installed twice or something. Either that or you are intentionally wasting our time to chase a non-issue. The log is showing you have something installed that does not belong there. We are not asking.

**Comment by xiaoxiao921** [comment]
![image](https://github.com/YimMenu/YimMenu/assets/837334/c7a534a7-c2a7-450d-b166-c8efde822268)
https://www.gta5-mods.com/maps/fib-mansion-rooftop-ymap-xml

**Comment by alyxme** [comment]
I found that file at "D:\Games\GTAV\Resources\XMLs\bb6fdd-FIB Mansion\" too. But there a bunch of other subfolders and archieves too. I used this "Resources\" as folder where I dump everything related, dlls, xmls, jsons by subfolders. No twice installation! I moved this folder back after reinstalling. Hope it will clear about the reason! Did that folder/path is used by yimmenu? Previously I moved it when trying find the solution to the cache issue but there was injector problem in first.
